### PR TITLE
 Migrate documentation for all string functions from scalar_functions.md to code 

### LINKED ISSUE #12774: Migrate documentation for all string functions from scalar_functions.md to code
### Is your feature request related to a problem or challenge?

As part of #12740 this ticket is to migrate all the existing documentation for string and unicode functions from the scalar_functions.md file to code and to remove from the scalar_functions.md file. In addition sql examples were added for all of the updated functions.

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Omega359**
## Which issue does this PR close?

Closes #12774 

## Rationale for this change

Documentation migration

## What changes are included in this PR?

Documentation related changes for string and unicode functions.

## Are these changes tested?

Yes.

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Thank you @Omega359  -- this is epic. 📖 

It seems to me like you have not only ported the documentation but have improved it significantly with new examples 

Many thanks again

File: docs/source/user-guide/sql/scalar_functions_new.md [hunk 1]
```diff
@@ -66,24 +66,650 @@ coalesce(expression1[, ..., expression_n])
...
 
 #### Arguments
 
-- **str**: String expression to operate on. Can be a constant, column, or function that evaluates to or can be coerced to a Utf8, LargeUtf8 or a Utf8View.
+- **str**: String expression to operate on. Can be a constant, column, or function, and any combination of operators.
+
+#### Example
+
+```sql
+> select ascii('abc');
```
[the remark this answers is not in the document]
Comment (Omega359) [reply docs/source/user-guide/sql/scalar_functions_new.md:123]: I had thought the same thing. I was contemplating adding a note saying that the 'ascii' name was for historical reasons but decided to just leave it for now :)

**Comment by alamb** [comment]
📖 !

**Comment by alamb** [comment]
Thanks again @Omega359

**Pull request merged**